src/com/oltpbenchmark/benchmarks/wikipedia/WikipediaWorker.java

```diff
@@ -62,58 +60,65 @@ private String generateUserIP() {
 
     @Override
     protected TransactionStatus executeWork(TransactionType nextTransaction) throws UserAbortException, SQLException {
-        WikipediaOperation t = null;
-        
+		Flat z_users = new Flat(this.rng(), 1, this.num_users);
+		Zipf z_pages = new Zipf(this.rng(), 1, this.num_pages, WikipediaConstants.USER_ID_SIGMA);
+
         Class<? extends Procedure> procClass = nextTransaction.getProcedureClass();
         boolean needUser = (procClass.equals(AddWatchList.class) ||
                             procClass.equals(RemoveWatchList.class) ||
-                            procClass.equals(GetPageAuthenticated.class));    
-        while (t == null) {
-            t = this.generator.nextTransaction();
-            if (needUser && t.userId == 0) {
-                t = null;
-            }
-        } // WHILE
-        assert(t != null);
-        if (t.userId != 0) t.userId = this.usersRng.nextInt();
-        
+                            procClass.equals(GetPageAuthenticated.class));
```

Comment:
Formatting! Andy **hates** tabs!

Reply:
(The original file was in tabs, so the diff is going to include all the code in the file now 😅 . Noted, will make the change!)